nav2_smac_planner/src/smac_planner_hybrid.cpp

```diff
@@ -337,6 +351,10 @@ nav_msgs::msg::Path SmacPlannerHybrid::createPlan(
   NodeHybrid::CoordinateVector path;
   int num_iterations = 0;
 
+  std::shared_ptr<std::vector<std::tuple<float, float>>> expansions = nullptr;
+  if (_viz_expansions) {
+    expansions = std::make_shared<std::vector<std::tuple<float, float>>>();
```

Comment:
Also, a unique pointer that you `.get()` on will reduce some compute time on reference counter semantics and pointer overhead. `.get()` will still be nullptr if its not allocated I believe

Reply:
> You don't seem to pass this to the function?

My bad, bad merge when switching from Humble to main.
Side note, I noticed tolerance is now set to 0 in: 
https://github.com/ros-planning/navigation2/blob/85fd06f20413e348059205a84ecb50765fc8f0c7/nav2_smac_planner/src/smac_planner_hybrid.cpp#L341
Is this voluntary and handled elsewhere ? Or does it means that we don't support planning with tolerance in the SmacHybrid planner ?




> Also, a unique pointer that you `.get()` on will reduce some compute time on reference counter semantics and pointer overhead. `.get()` will still be nullptr if its not allocated I believe

Done

Reply:
The feasible planners don't have margin for error since "close enough" can't know if its kinematically achievable. anyway, the analytic expansions are what's getting the precise path to goal once within the general window of the goal so it should be exact if its possible at all.